src/os/bluestore/BlueStore.cc

```diff
@@ -8666,21 +8664,15 @@ void BlueStore::_deferred_submit_unlock(OpSequencer *osr)
 void BlueStore::_deferred_aio_finish(OpSequencer *osr)
 {
   dout(10) << __func__ << " osr " << osr << dendl;
-  assert(osr->deferred_running);
-  DeferredBatch *b = osr->deferred_running;
+  DeferredBatch *b;
 
   {
     std::lock_guard<std::mutex> l(deferred_lock);
-    assert(osr->deferred_running == b);
-    osr->deferred_running = nullptr;
-    if (!osr->deferred_pending) {
+    b = osr->deferred_running.front();
```

Comment:
My original fix did something like that, where we retry deferred_try_submit, but it was very awkward and relied on fragile Throttler changes.  Simply allowing multiple batches to be in flight at once seems much simpler.  See https://github.com/liewegas/ceph/commits/wip-20295-rebased

Fixed the aio completion assumption!